### ISSUE TITLE
🐛 [firebase_remote_config] set defaults with Dart boolean does not work

### ISSUE THREAD

**Issue by stephenhuey** [FirebaseExtended/flutterfire#3797]
## Bug report

I assume this is a bug since some docs out there like this one imply that you should be able to pass an actual boolean among the defaults for initializing RemoteConfig:

[https://firebase.google.com/docs/reference/android/com/google/firebase/remoteconfig/FirebaseRemoteConfig#setDefaultsAsync(int)](https://firebase.google.com/docs/reference/android/com/google/firebase/remoteconfig/FirebaseRemoteConfig#setDefaultsAsync(int))

**Describe the bug**
Not sure if this changed or not but I was tearing my hair out for a while assuming the problem was in other places.  Essentially, the map I passed to my RemoteConfig instance with `setDefaults` had some boolean values for some of the keys, like so:

```
Map<String, dynamic> map = {
      'myKey': true,
}
```

However, when I called `getBool('myKey')` to find out whether it was true or false, it was always false even when I was certain I was setting it to true.  I assumed there was an exception somewhere during initialization, but then I debugged into RemoteConfig itself and realized what was going on.  That `getBool` function calls this next one in `remote_config_value.dart`:

```
  bool asBool() {
    if (_value != null) {
      final String strValue = const Utf8Codec().decode(_value);
      return strValue.toLowerCase() == 'true';
    } else {
      return RemoteConfig.defaultValueForBool;
    }
  }
```

My breakpoint on the 4th line showed me that RemoteConfig was internally storing the value for true as a 1 and so `strValue` there is `"1"` and of course that does not equal `"true"` so I always got the default boolean which was false.  But if I set my value in my defaults to a String `"true"` then getBool always returns what I expect for the key.    **Is this the way it is intended to work?**  Again, I would think that link at the top would imply that we can pass actual Dart bool values, not just String ones (I know that is not a link specific to Dart/Flutter but the Dart/Flutter docs are pretty sparse and I would assume it would work the same way everywhere).  

My pubspec.yaml has `firebase_remote_config: "^0.4.0"` and I just noticed my pubspec.lock has `version: "0.4.0+1"` right now.  

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.20.4, on Mac OS X 10.15.7 19H2, locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.49.0)
[✓] Connected device (2 available)

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.9.2
Flutter SDK 1.20.4
myapp 1.0.0+2

dependencies:
- cloud_firestore 0.14.1+2 [flutter meta quiver firebase_core firebase_core_platform_interface cloud_firestore_platform_interface cloud_firestore_web]
- cloud_functions 0.6.0+1 [meta flutter firebase_core cloud_functions_platform_interface cloud_functions_web]
- cupertino_icons 0.1.3
- device_info 0.4.2+8 [flutter device_info_platform_interface]
- enum_to_string 1.0.11
- firebase_analytics 6.0.1 [meta flutter firebase_core firebase_analytics_web firebase_analytics_platform_interface]
- firebase_auth 0.18.1+1 [meta firebase_core firebase_core_platform_interface firebase_auth_platform_interface firebase_auth_web flutter]
- firebase_core 0.5.0 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- firebase_dynamic_links 0.6.0+1 [flutter firebase_core]
- firebase_messaging 7.0.2 [meta flutter firebase_core]
- firebase_remote_config 0.4.0+1 [flutter firebase_core]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_typeahead 1.8.7 [flutter flutter_keyboard_visibility]
- intl 0.16.1 [path]
- network_to_file_image 2.3.3 [flutter]
- overlay_support 1.0.5 [flutter async]
- path_provider 1.6.11 [flutter path_provider_platform_interface path_provider_macos path_provider_linux]
- provider 4.1.2 [flutter nested collection]
- purchases_flutter 1.3.1 [flutter]
- sembast 2.4.7+6 [meta path synchronized]
- share 0.6.5+2 [meta mime flutter]
- simple_connectivity 0.1.1 [flutter meta]
- timeago 2.0.27

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.4.2 [collection]
- boolean_selector 2.0.0 [source_span string_scanner]
- characters 1.0.0
- charcode 1.1.3
- clock 1.0.1 [meta]
- cloud_firestore_platform_interface 2.1.1 [flutter meta collection firebase_core plugin_platform_interface]
- cloud_firestore_web 0.2.0+3 [flutter flutter_web_plugins firebase http_parser meta firebase_core cloud_firestore_platform_interface js]
- cloud_functions_platform_interface 2.0.0 [flutter meta firebase_core plugin_platform_interface]
- cloud_functions_web 2.0.0 [cloud_functions_platform_interface flutter flutter_web_plugins firebase http_parser meta]
- collection 1.14.13
- device_info_platform_interface 1.0.1 [flutter meta plugin_platform_interface]
- fake_async 1.1.0 [clock collection]
- file 5.2.1 [intl meta path]
- firebase 7.3.0 [http http_parser js]
- firebase_analytics_platform_interface 1.0.3 [flutter meta]
- firebase_analytics_web 0.1.1 [flutter flutter_web_plugins firebase firebase_analytics_platform_interface meta]
- firebase_auth_platform_interface 2.1.0 [flutter meta firebase_core plugin_platform_interface]
- firebase_auth_web 0.3.1 [flutter flutter_web_plugins firebase meta http_parser intl firebase_core firebase_auth_platform_interface js]
- firebase_core_platform_interface 2.0.0 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.0 [firebase firebase_core_platform_interface flutter flutter_web_plugins meta js]
- flutter_keyboard_visibility 3.2.2 [flutter]
- flutter_web_plugins 0.0.0 [flutter characters collection meta typed_data vector_math]
- http 0.12.2 [http_parser path pedantic]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- js 0.6.2
- matcher 0.12.8 [stack_trace]
- meta 1.1.8
- mime 0.9.7
- nested 0.0.4 [flutter]
- path 1.7.0
- path_provider_linux 0.0.1+2 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 0.0.4+4 [flutter]
- path_provider_platform_interface 1.0.3 [flutter meta platform plugin_platform_interface]
- pedantic 1.9.0
- platform 2.2.1
- plugin_platform_interface 1.0.3 [meta]
- process 3.0.13 [file intl meta path platform]
- quiver 2.1.3 [matcher meta]
- sky_engine 0.0.99
- source_span 1.7.0 [charcode collection meta path term_glyph]
- stack_trace 1.9.5 [path]
- stream_channel 2.0.0 [async]
- string_scanner 1.0.5 [charcode meta source_span]
- synchronized 2.2.0+2
- term_glyph 1.1.0
- test_api 0.2.17 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.2.0 [collection]
- vector_math 2.0.8
- xdg_directories 0.1.0 [path process flutter]
```

</details>

---

**Comment by TahaTesser** [comment]
Hi @stephenhuey 
Can you please provide  a complete reproducible minimal code sample
Thank you

**Comment by stephenhuey** [comment]
> Hi @stephenhuey
> Can you please provide a complete reproducible minimal code sample
> Thank you

Thanks for your reply.  I just tried to create a minimally viable example that fails in the same way, and even after trying several recent package versions of firebase_remote_config I was still not able to replicate my issue.  I don't think I was dealing with a race condition in my big project, but all I know is that I was tearing my hair out for a long time before I gave in and made all the config default values String instead of bool.  Sorry for taking up your time, but for whatever reason, getBool works fine in my minimal example no matter whether I set the default values with String or bool variables.
